src/components/App/App.tsx

```diff
@@ -52,13 +64,55 @@ export default class App extends React.Component<{}, AppState> {
         console.log('Selected movie id is ' + selectedMovieId);
     };
 
+    onFormSubmit = (movie: Movie) => {
+        console.log('Submitted movie id is ' + movie.title);
+    };
+
+    onCloseDialog = () => {
+        this.setState({ showModal: false });
+    };
+
     getSelectedMovieById = (movieId: number): Movie | undefined => {
         return this.state.movies.find(({ id }) => id === movieId);
     };
 
+    onDeleteMovie = () => {
+        console.log('Movie with id' + this.state.selectedMovieId + 'was deleted');
+    };
+
+    renderModal() {
+        if (!this.state.showModal) return null;
+        let dialogContent = null;
+        let dialogTitle = '';
+
+        if (this.state.isAddMovieModal) {
+            dialogContent = <MovieForm onSubmit={this.onFormSubmit} />;
+            dialogTitle = 'Add Movie';
```

Comment:
Then here we can use map. For example 
`const DIALOG_TITLE_MAP = {
  Add: 'Add Movie',
  Delete: 'Delete movie'
}`
And use it like 
`
  dialogContent = DIALOG_MOVIE_MAP[this.state.movieModalType]
`

Reply:
fixed